src/services/bunker_cases/abnormal_cl_rebase.py

```diff
@@ -21,187 +22,182 @@
-        )
 
-        if nearest_slot < distant_slot:
-            raise ValueError(f"{nearest_slot=} should be less than {distant_slot=} in specific CL rebase calculation")
-        if distant_slot < self.last_report_ref_slot:
-            raise ValueError(
-                f"{distant_slot=} should be greater than {self.last_report_ref_slot=} in specific CL rebase calculation"
-            )
-
-        nearest_blockstamp = get_first_non_missed_slot(
-            self.w3.cc,
-            SlotNumber(nearest_slot),
-            last_finalized_slot_number=ref_blockstamp.slot_number,
-            ref_epoch=EpochNumber(nearest_slot // self.c_conf.slots_per_epoch),
-        )
-
-        distant_blockstamp = get_first_non_missed_slot(
-            self.w3.cc,
-            SlotNumber(distant_slot),
-            last_finalized_slot_number=ref_blockstamp.slot_number,
-            ref_epoch=EpochNumber(distant_slot // self.c_conf.slots_per_epoch),
-        )
+        nearest_blockstamp, distant_blockstamp = self._get_nearest_and_distant_blockstamps(blockstamp)
 
         if nearest_blockstamp.block_number == distant_blockstamp.block_number:
             logger.info(
                 {"msg": "Nearest and distant blocks are the same. Specific CL rebase will be calculated once"}
             )
-            specific_cl_rebase = self._calculate_cl_rebase_between(nearest_blockstamp, ref_blockstamp)
+            specific_cl_rebase = self._calculate_cl_rebase_between_blocks(nearest_blockstamp, blockstamp)
             logger.info({"msg": f"Specific CL rebase: {specific_cl_rebase} Gwei"})
             return specific_cl_rebase < 0
 
-        nearest_cl_rebase = self._calculate_cl_rebase_between(nearest_blockstamp, ref_blockstamp)
+        nearest_cl_rebase = self._calculate_cl_rebase_between_blocks(nearest_blockstamp, blockstamp)
         logger.info({"msg": f"Nearest specific CL rebase {nearest_cl_rebase} Gwei"})
         if nearest_cl_rebase < 0:
             return True
-        distant_cl_rebase = self._calculate_cl_rebase_between(distant_blockstamp, ref_blockstamp)
+
+        distant_cl_rebase = self._calculate_cl_rebase_between_blocks(distant_blockstamp, blockstamp)
         logger.info({"msg": f"Distant specific CL rebase {distant_cl_rebase} Gwei"})
         if distant_cl_rebase < 0:
             return True
+
         return False
 
-    def _calculate_cl_rebase_between(self, prev_blockstamp: ReferenceBlockStamp, ref_blockstamp: ReferenceBlockStamp) -> Gwei:
+    def _get_nearest_and_distant_blockstamps(
+        self, blockstamp: ReferenceBlockStamp
+    ) -> tuple[ReferenceBlockStamp, ReferenceBlockStamp]:
+        """Get nearest and distant blockstamps. Calculation including missed slots"""
+        nearest_slot = SlotNumber(
+            blockstamp.slot_number - self.b_conf.rebase_check_nearest_epoch_distance * self.c_conf.slots_per_epoch
```

Comment:
`blockstamp.slot_number` -> `blockstamp.ref_slot` since we are operating with time here

Reply:
fixed